### PR TITLE
Add StringMap and StringSet

### PR DESCRIPTION
**Pull request by arv**
Use those instead of Object.create(null).

### CODE REVIEW

**Comment by arv** [comment]
@johnjbarton

File: src/util/StringSet.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  add(value) {
+    assertString(value);
+    this.storage_[value] = true;
+  }
+
+  has(value) {
+    assertString(value);
+    return this.storage_[value] !== undefined;
+  }
+  delete(value) {
```
Comment (johnjbarton) [inline src/util/StringSet.js]: Consistent spacing between methods would be nice to have.

Comment (arv) [inline src/util/StringSet.js]: Agreed. Done.

**Comment by johnjbarton** [comment]
LGTM The commit description could explain why this is desirable, esp. I suppose we will want future commits to avoid `Object.create(null)` in favor of StringSet/Map.

**Comment by arv** [comment]
Thanks.

I'll squash and update the description.

**Pull request merged**